rails/app/controllers/admin/project_links_controller.rb

```diff
@@ -0,0 +1,85 @@
+class Admin::ProjectLinksController < ApplicationController
+  include RestrictedController
+
+  before_filter :check_for_project
+  before_filter :get_scoped_projects, only: ['new', 'edit']
+  before_filter :find_project_link, only: ['show', 'edit', 'update', 'destroy']
+
+
+  def check_for_project
+    return unless params[:project_id]
+
+    @project = Admin::Project.find(params[:project_id])
+    authorize @project
```

Comment:
This is kind of a redundant check except in the case of index, and new.
In other cases the project_link object is checked too and that check is more important. 
I think a good way to clarify this is to remove this check here, and move this authorize check directly to the index and new actions. like `authorize @project if @project`